### PR TITLE
Destination Postgres: CDK T+D initial state gathering

### PR DESCRIPTION
**Pull request by gisripa**
## What
Adapting to CDK changes in #35342

### CODE REVIEW

**Comment by gisripa** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35385?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* `master`
  * **#35342** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35342?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **#35385** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35385?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **#35376** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35376?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **#35354** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35354?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresDestinationHandler.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+public class PostgresDestinationHandler extends JdbcDestinationHandler {
+
+  public PostgresDestinationHandler(String databaseName, JdbcDatabase jdbcDatabase) {
+    super(databaseName, jdbcDatabase);
+  }
+
+  @Override
+  protected String toJdbcTypeName(AirbyteType airbyteType) {
+    // This is mostly identical to the postgres implementation, but swaps jsonb to super
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresDestinationHandler.java:24]: I don't think this comment applies :P

File: airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresSqlGeneratorIntegrationTest.java [hunk 2]
```diff
@@ -96,29 +95,11 @@ public void testCreateTableIncremental() throws Exception {
...
-        () -> assertEquals("timetz", existingTable.get().columns().get("time_with_timezone").type()),
-        () -> assertEquals("time", existingTable.get().columns().get("time_without_timezone").type()),
-        () -> assertEquals("date", existingTable.get().columns().get("date").type()),
-        () -> assertEquals("jsonb", existingTable.get().columns().get("unknown").type()));
-    // TODO assert on table indexing, etc.
+    List<DestinationInitialState> initialStates = destinationHandler.gatherInitialState(List.of(incrementalDedupStream));
+    assertEquals(1, initialStates.size());
+    final DestinationInitialState initialState = initialStates.getFirst();
+    assertTrue(initialState.isFinalTablePresent());
+    assertFalse(initialState.isSchemaMismatch());
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresSqlGeneratorIntegrationTest.java:102]: can we do an assertEquals on the entire initialState object? In theory we can hand-construct the expected value, right?

**Review by edgao — APPROVED** [review]

File: airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresSqlGeneratorIntegrationTest.java (hunk 2 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresSqlGeneratorIntegrationTest.java:102]: owait, we don't have the table schema at all anymore right?

File: airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresDestinationHandler.java (hunk 1 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresDestinationHandler.java:24]: lol it was inverted in redshift's handler, i fixed there and propagated that here 😂

**Pull request merged**